### PR TITLE
Added a first draft of CONTRIBUTING.md

### LINKED ISSUE #19: Community standards: Add CONTRIBUTING.md
It can be largely based on the existing documentation ([developer_setup.md](https://github.com/ImperialCollegeLondon/virtual_rainforest/blob/develop/docs/source/development/developer_setup.md "developer_setup.md") and [code_development_strategy.md](https://github.com/ImperialCollegeLondon/virtual_rainforest/blob/develop/docs/source/development/code_development_strategy.md "code_development_strategy.md")), but in a more condensed format

More on Community Standards: [Community Standards (github.com)](https://github.com/ImperialCollegeLondon/virtual_rainforest/community)

### PR DESCRIPTION
**Pull request by davidorme**
Closes #19

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Contributing to the Virtual Rainforest
+
+We're really happy that you are thinking about contributing to the Virtual Rainforest
+project. The whole point of the project is to generate a tool that can be used widely by
+the community and the best way for that to happen is for the community to build it.
+
+To make contributing as seemless as possible, please note these developer guidelines.
```
Comment (alexdewar) [inline CONTRIBUTING.md:7]: Spelling:
```suggestion
To make contributing as seamless as possible, please note these developer guidelines.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Development setup
+
+Further notes are available [here](source/docs/develop/developer_setup.md) but - once
+you have cloned the Virtual Rainforest repository to your own machine - you will need to
+set up our development toolchain on your machine to contribute to the project.
+
+* We use `poetry` ([](https://https://python-poetry.org/)) to manage the package
```
Comment (alexdewar) [inline CONTRIBUTING.md:15]: You *could* make this an inline hyperlink:
```suggestion
* We use ([poetry](https://https://python-poetry.org/)) to manage the package
```

Same for other links. This is a stylistic choice though so up to you

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  development. Once you have installed `poetry`, you can use `poetry install` within
+  the repo to start using it.
+* We use `pre-commit` ([](https://pre-commit.com/)) to maintain code quality on
```
Comment (alexdewar) [inline CONTRIBUTING.md:18]: See above

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  submitted code. Before changing the code, install `pre-commit` and then use
+  `pre-commit` install to make sure that your code is being checked.
+
+## Contributing code
+
+* Code contributed to the Virtual Rainforest **must** address a documented issue on the
+  [issue tracker](https://github.com/ImperialCollegeLondon/virtual_rainforest/issues).
```
Comment (alexdewar) [inline CONTRIBUTING.md:25]: I think this requirement is a bit too hard: a user may want to contribute a small one-line fix and making them jump through extra hoops might be offputting.

Again though, it's your project, so up to you!

**Review by alexdewar — CHANGES_REQUESTED** [review]
I've made a couple of suggestions.

LGTM otherwise though except for one spelling mistake.

**Comment by alexdewar** [comment]
Another small suggestion: I'd tell contributors that they are expected to abide by your code of conduct and link to the document.

**Comment by alexdewar** [comment]
Also, while I've given my review it'd also be great to have someone else in your team give their own review, so let's hold off on merging until @LivDaniel has given her two cents.

File: CONTRIBUTING.md (hunk 2 above)
Comment (davidorme) [reply CONTRIBUTING.md:15]: I guess I think it is useful for code tools to be styled - it tells people they are programs - and having a link on a code styled word can be less visible. Testing [`here`](https://httpstat.us/200).

Comment (alexdewar) [reply CONTRIBUTING.md:15]: I hear you. As I say, up to you!

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+* Obviously, that code needs to pass the `pre-commit` checks!
+* Now submit a **pull request** to merge your branch into the `develop` branch of the
+  Virtual Rainforest project.
+* We will then review the contributed code and merge it once any problems have been
+  resolved.
+
+## The `pytest` framework
+
+We use `pytest` ([](https://docs.pytest.org/)) to run continuous integration and other
+testing on the code in Virtual Rainforest. If you are adding new functionality or
```
Comment (LivDaniel) [inline CONTRIBUTING.md:44]: in the Virtual Rainforest

File: CONTRIBUTING.md (hunk 2 above)
Comment (davidorme) [reply CONTRIBUTING.md:15]: That's not too bad as rendered in GitHub.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -4,25 +4,30 @@ We're really happy that you are thinking about contributing to the Virtual Rainf
...
 set up our development toolchain on your machine to contribute to the project.
 
-* We use `poetry` ([](https://https://python-poetry.org/)) to manage the package
+* We use [`poetry`](https://https://python-poetry.org/) to manage the package
   development. Once you have installed `poetry`, you can use `poetry install` within
   the repo to start using it.
-* We use `pre-commit` ([](https://pre-commit.com/)) to maintain code quality on
+* We use [`pre-commit`](https://pre-commit.com/) to maintain code quality on
   submitted code. Before changing the code, install `pre-commit` and then use
   `pre-commit` install to make sure that your code is being checked.
```
Comment (LivDaniel) [inline CONTRIBUTING.md:20]: Should it be 'pre-commit install'? as above you had ' poetry install'

**Review by alexdewar — APPROVED** [review]
Looks like you've addressed everything :smile:

**Review by LivDaniel — APPROVED** [review]

**Pull request merged**